coffea/processor/executor.py

```diff
@@ -555,30 +623,152 @@ def __call__(
         if self.compression is not None:
             function = _compression_wrapper(self.compression, function)
 
-        def processwith(pool):
-            gen = _futures_handler(
-                {pool.submit(function, item) for item in items}, self.tailtimeout
+        def processwith(pool, mergepool):
+            reducer = _reduce(self.compression)
+            FH = FuturesHolder(
+                set(pool.submit(function, item) for item in items), refresh=2
             )
+
+            nparts, minred, maxred = 5, 4, 100
+            if isinstance(self.merging, tuple):
+                if len(self.merging) == 3:
+                    nparts, minred, maxred = self.merging
```

Comment:
Move above `pool.submit` and return an error if the format isn't valid

Reply:
Best would be a dataclass `__post_init__` hook https://docs.python.org/3/library/dataclasses.html#post-init-processing